### PR TITLE
fix witness metadata on invalid reports

### PR DESCRIPTION
**Pull request by jeffgrunewald**
At the stage at the end of `do_witness_verifications` in the iot_verifier poc.rs module where we assign non-zero values to invalid witness metadata we should have validated the witness's on-chain metadata at least (the invalid witness report was invalid for a different reason) so we should assign the witnesses's metadata to its section of the invalid report rather than the beaconer's.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,6 +22,7 @@ members = [
     "solana",
     "task_manager",
 ]
+resolver = "2"
```
Comment (madninja) [inline Cargo.toml:25]: the changes in this file seem unneeded

Comment (jeffgrunewald) [reply Cargo.toml:25]: The latest rust (1.72.0) has started throwing some noisy warnings/notes on compile without this

**Review by bbalser — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (madninja) [reply Cargo.toml:25]: how incredibly annoying that they just randomly started requiring explicitly using a next gen resolver.. oh well

Comment (jeffgrunewald) [reply Cargo.toml:25]: They didn’t, just required you explicitly declare one. If this should “1” that’s cool too

**Review by andymck — APPROVED** [review]
ffs, couldnt believe this one when i saw it

**Pull request merged**